test/cpp/test_xla_sharding.cpp

```diff
@@ -83,6 +83,13 @@ TEST_F(XLAShardingTest, ShardTensor) {
   EXPECT_EQ(shards.size(), 8);
   EXPECT_EQ(shards[0].sizes(), c10::ArrayRef<long>({1, 8, 2, 2}));
   EXPECT_EQ(shards[7].sizes(), c10::ArrayRef<long>({1, 8, 1, 2}));
+
+  // 4D tiled and padded, all shard sizes should be idential.
+  shards =
+      ShardingUtil::ShardTensor(tensor, sharding, devices, /*padded=*/true);
```

Comment:
in what case should we pad?

Reply:
We should pad always -- I was debating whether to pad outside this util function, leaving it more generic tensor sharding. Instead, I introduced a padded (default true) arg for the util function.